packages/react/src/analytics/integrations/AnalyticsApi/AnalyticsApi.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  type AnalyticsApiIntegrationOptions,
+  type AnalyticsApiTrackData,
+} from './types/types.js';
+import { client } from '@farfetch/blackout-client';
+import {
+  type EventData,
+  integrations,
+  type LoadIntegrationEventData,
+  type StrippedDownAnalytics,
+  type TrackTypesValues,
+  utils,
+} from '@farfetch/blackout-analytics';
+import { isArray } from 'lodash';
+
+export default class AnalyticsAPI extends integrations.Integration<AnalyticsApiIntegrationOptions> {
+  static override [utils.CONSENT_CATEGORIES_PROPERTY] =
+    utils.DefaultConsentKeys.MARKETING;
+  private debugMode: boolean | undefined;
+
+  /**
+   * Creates an instance of Analytics Api integration.
+   *
+   * @param options               - User configured options.
+   * @param loadData              - Analytics's load event data.
+   * @param strippedDownAnalytics - Analytics stripped down instance.
+   */
+  constructor(
+    options: AnalyticsApiIntegrationOptions,
+    loadData: LoadIntegrationEventData,
+    strippedDownAnalytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, strippedDownAnalytics);
+
+    // Check if options don't exists
+    if (!options) {
+      throw new Error(
+        '[Analytics API] - Analytics Api integration options must be set.',
+      );
+    }
+
+    if (!isArray(options.conversionsAPI?.eventsList)) {
+      throw new Error(
+        '[Analytics API] - The value `conversionsAPI.eventsList` from Analytics Api integration options must be an array.',
+      );
+    }
+
+    this.debugMode =
+      utils.getCookie('analyticsAPIDebug')?.toLowerCase() === 'true';
+  }
+
+  /**
+   * Track method that will call Analytics API with the necessary data per event.
+   * If there's more than one pageType per event, it will call Analytics API with each page type at a time.
+   *
+   * @param data - Track event data.
+   *
+   * @returns Promise that will resolve when the method finishes.
+   */
+  override async track(data: EventData<TrackTypesValues>) {
+    if (this.debugMode) {
+      utils.logger.info('ANALYTICS API track:', {
+        data,
+        options: this.getOptions(),
+      });
+    }
+
+    if (
+      (this.options.conversionsAPI.eventsList as Array<string>).includes(
+        data.event,
+      )
+    ) {
+      return await this.callAPI({
+        data,
+        options: this.getOptions(),
+      });
+    }
+  }
+
+  /**
+   * Calls the Analytics API with the given payload.
+   *
+   * @param data - Analytics' event data.
+   */
+  async callAPI(data: AnalyticsApiTrackData) {
+    await client.post('/analytics/v1/events', data);
```

Comment:
Shouldn't we create a client in `@farfetch/blackout-client` and use it as we do with Omnitracking? Also, it would be interesting to allow the user to specify the client he wants and only use the default if he does not specify anything.

Reply:
I'm not sure if this case are already cover, but this initiative has some improvements in future, due delivery time. I think we could postpone this as well, and create an task for latter.